preprocessing/data_preprocessor.py

```diff
@@ -122,12 +102,91 @@ def segment_nifti_files(self, nifti_file_path: str) -> None:
         """
         out_path = os.path.join(self.segment_dir, nifti_file_path.split(
             '/')[-1].replace('.nii.gz', ''))
-        totalsegmentator(nifti_file_path, out_path, roi_subset=[
-                         'liver', 'spleen', 'kidney_right', 'kidney_left', 'small_bowel'])
+        totalsegmentator(nifti_file_path, out_path, roi_subset=ORGANS)
+
+    def json_organs(self, subject) -> None:
+        value = {subject: {}}
+
+        transform = CropForeground(return_coords=True)
+
+        for organ in ORGANS:
+            mask, _ = load_nifti(os.path.join(self.segment_dir, subject, organ + ".nii.gz"))
+            _, coords_in, coords_end = transform(np.expand_dims(mask, axis=0))
+
+            coords_in = coords_in.tolist()
+            coords_end = coords_end.tolist()
+
+            value[subject][organ] = {
+                'start': coords_in,
+                'end': coords_end
+            }
+
+        # Check if the JSON file exists
+        filename = os.path.join(self.working_dir, 'cropped_organs.json')
+        if os.path.exists(filename):
+            with open(filename, 'r') as json_file:
+                existing_data = json.load(json_file)
+        else:
+            existing_data = {}
+
+        # Update the existing data with the new subject's data
+        existing_data.update(value)
+
+        # Save the updated dictionary to JSON
+        with open(filename, 'w') as json_file:
+            json.dump(existing_data, json_file, indent=6)
+    
+    def crop_organs(self, subject):
+        filename = os.path.join(self.working_dir, 'cropped_organs.json')
+        with open(filename, 'r') as json_file:
+            coordinates = json.load(json_file)
+
+        os.makedirs(os.path.join(
+            self.nifti_dir,
+            subject.strip(".nii.gz")), exist_ok=True)
+        
+        img, _ = load_nifti(os.path.join(self.nifti_dir, subject))
+
+        for organ in ORGANS:
+            organ_coordinates = coordinates[subject.strip(".nii.gz")][organ]
+            organ_img = img[
+                organ_coordinates["start"][0]:organ_coordinates["end"][0],
+                organ_coordinates["start"][1]:organ_coordinates["end"][1],
+                organ_coordinates["start"][2]:organ_coordinates["end"][2]
+            ]
+            organ_img = nib.Nifti1Image(organ_img, np.eye(4))
+
+            nib.save(organ_img, os.path.join(self.nifti_dir,
+                                            subject.strip(".nii.gz"),
+                                            organ + ".nii.gz"))
+            
+        os.rename(os.path.join(self.nifti_dir, 
+                               subject), 
+                  os.path.join(self.nifti_dir,
+                               subject.strip(".nii.gz"),
+                               subject))
 
+    def __call__(self) -> None:
```

Comment:
@FrancescVP the call method shoud be placed after the init method, rather than at the end of the script ! <3